### ISSUE TITLE
prevent players to use the `[bot]` icon at start of their name

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/Unvanquished#2125]
This is a really minor detail, hence the milestone, and I have to say I never seen anyone use that to really troll people.

Bots have, by default, a "chip" icon at the beginning of their name. This is only for automatic names, but is usually followed by admins AFAICT, but nothing prevents a player to do the same, adding to the confusion.
I wonder if it would be ok to prevent players to use the conventions that signal bots.

Honestly, this really is a minor thing, and it could be closed as wontfix, no problem.

**Comment by sweet235** [comment]
> I have to say I never seen anyone use that to really troll people
I have.

**Comment by DolceTriade** [comment]
Seems like a reasonable idea configurable with a cvar.

**Comment by illwieckz** [comment]
We would have to check the name prefix after removing color codes to catch all combinations.

**Comment by DolceTriade** [comment]
Do emojis work with color codes?

**Comment by necessarily-equal** [comment]
So what I'm doing in https://github.com/Unvanquished/Unvanquished/pull/2276 is to forbid the computer chip entirely in players' names. So I guess it resists to color codes, but also is more restrictive.

If somebody else does want to try implementing something, be wary that [bOT] is also a valid way of writting the emoticon.

**Comment by illwieckz** [comment]
> If somebody else does want to try implementing something, be wary that [bOT] is also a valid way of writting the emoticon.

This is probably an undocumented feature you should not rely on. Assume it's because it's implemented like `with x in [x] do /emoticons/x` and it happens that the pkzip implementation of the VFS is case insensitive.

Something working not means something is valid, if that breaks, that would not be a bug.

**Comment by necessarily-equal** [comment]
> This is probably an undocumented feature you should not rely on

Okok, but I don't think a griever is afraid of relying on an undocumented feature. Point is, we want to protect about `[bOT]` and variations too, because they work.

**Comment by illwieckz** [comment]
Yes, I assume a complete way to implement it is to get the non-color-coded variant of the string, lowercase it, and check for `^\[bot\]`.

**Comment by illwieckz** [comment]
well, maybe check for leading spaces too.

**Comment by necessarily-equal** [comment]
The problem is that "checking for leading $x" is not possible because 
we are using unicode. There are many non-printable that are intended, 
or that can used in a way they won't be rendered.

**Comment by slipher** [comment]
It might make sense to implement this by integrating with the `g_emoticonsAllowedInNames` code. IIRC it works by overwriting one of the square brackets.
